src/main/portfolio_driver.cpp

```diff
@@ -0,0 +1,531 @@
+  }
+
+  uint64_t total_timeout = ctx.solver().getOptionInfo("tlimit").uintValue();
+  if (total_timeout == 0)
+  {
+    total_timeout = 1200;
+  }
+
+  PortfolioProcessPool pool(ctx, ctx.parseCommands(d_parser));
+
+  return pool.run(strategy);
+#else
+  Warning() << "Can't run portfolio without <sys/wait.h>.";
+  return ctx.solveContinuous(d_parser, false);
+#endif
+}
+
+std::ostream& operator<<(std::ostream& os, const PortfolioConfig& config)
+{
+  for (const auto& o : config.d_options)
+  {
+    os << o.first << "=" << o.second << " ";
+  }
+  os << "timeout=" << config.d_timeout;
+  return os;
+}
+
+PortfolioStrategy PortfolioDriver::getStrategy(const std::string& logic)
+{
+  PortfolioStrategy s;
+  if (logic == "QF_LRA")
+  {
+    s.add(0.2)
+        .set("miplib-trick")
+        .set("miplib-trick-subs", "4")
+        .set("use-approx")
+        .set("lemmas-on-replay-failure")
+        .set("replay-early-close-depth", "4")
+        .set("replay-lemma-reject-cut", "128")
+        .set("replay-reject-cut", "512")
+        .set("unconstrained-simp")
+        .set("use-soi");
+    s.add()
+        .set("restrict-pivots", "false")
+        .set("use-soi")
+        .set("new-prop")
+        .set("unconstrained-simp");
+  }
+  else if (logic == "QF_LIA")
+  {
+    s.add()
+        .set("miplib-trick")
+        .set("miplib-trick-subs", "4")
+        .set("use-approx")
+        .set("lemmas-on-replay-failure")
+        .set("replay-early-close-depth", "4")
+        .set("replay-lemma-reject-cut", "128")
+        .set("replay-reject-cut", "512")
+        .set("unconstrained-simp")
+        .set("use-soi")
+        .set("pb-rewrites")
+        .set("ite-simp")
+        .set("simp-ite-compress");
+  }
+  else if (logic == "QF_NIA")
+  {
+    s.add(0.35).set("nl-ext-tplanes").set("decision", "justification");
+    s.add(0.05).set("nl-ext-tplanes").set("decision", "internal");
+    s.add(0.05).set("nl-ext-tplanes").set("decision", "justification-old");
+    s.add(0.05).set("nl-ext-tplanes", "false").set("decision", "internal");
+    s.add(0.05)
+        .set("arith-brab", "false")
+        .set("nl-ext-tplanes")
+        .set("decision", "internal");
+    s.add(0.25).set("solve-int-as-bv", "2").set("bitblast", "eager");
+    s.add(0.25).set("solve-int-as-bv", "4").set("bitblast", "eager");
+    s.add(0.25).set("solve-int-as-bv", "8").set("bitblast", "eager");
+    s.add(0.25).set("solve-int-as-bv", "16").set("bitblast", "eager");
+    s.add(0.5).set("solve-int-as-bv", "32").set("bitblast", "eager");
+    s.add().set("nl-ext-tplanes").set("decision", "internal");
+  }
+  else if (logic == "QF_NRA")
+  {
+    s.add(0.5).set("decision", "justification");
+    s.add(0.25)
+        .set("decision", "internal")
+        .set("nl-cov", "false")
+        .set("nl-ext", "full")
+        .set("nl-ext-tplanes");
+    s.add().set("decision", "internal").set("nl-ext", "none");
+  }
+  return s;
+}
+
+}  // namespace cvc5::main
```

Comment:
Should we already add the remaining competition configs here?

Reply:
Yeah, I can do that. The next block would have been a strategy for ~20 logics, and I was pondering how to make the `if (logic == "..." || logic == "..." || ...)` less horrible in this case.

Reply:
Since the comment in the competition script says:
```
# all logics with UF + quantifiers should either fall under this or special cases below
```
you could check whether the logic starts with A or contains UF and check the other quantified logics before that?

Reply:
@ajreynol opinions?

Reply:
Added the remaining configs.

Reply:
For now, I added a simple utility that allows for `isOneOf(logic, "foo", "bar", "baz", ...)`.